packages/components/src/modal/Modal.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Modal from './Modal';
+
+function makeModal({
+  className,
+  children,
+  keyboard,
+  isOpen,
+  centered,
+  onOpened,
+  toggle,
+  'data-testid': dataTestId,
+}: {
+  className?: string;
+  children?;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  onOpened?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}) {
+  return (
+    <Modal
+      className={className}
+      keyboard={keyboard}
+      isOpen={isOpen}
+      centered={centered}
+      onOpened={onOpened}
+      toggle={toggle}
+    >
+      {children}
+    </Modal>
+  );
+}
+
+it('mounts', () => {
+  render(makeModal({}));
+});
+
+it('does not render when isOpen is false', () => {
+  render(makeModal({ isOpen: false }));
+  expect(screen.queryByRole('dialog')).toBeNull();
+});
+
+it('renders when isOpen is true', async () => {
+  render(makeModal({ isOpen: true }));
+  const dialogs = screen.getAllByRole('dialog');
+  expect(dialogs.length).not.toBe(0);
```

Comment:
There should only be one `dialog`, clean up Modal.tsx so you can update all these tests to just be `expect(screen.queryByRole('dialog')).toBeNull()` or `expect(screen.queryByRole('dialog')).not.toBeNull()`